beacon-chain/p2p/encoder/ssz.go

```diff
@@ -99,7 +99,10 @@ func (e SszNetworkEncoder) doDecode(b []byte, to interface{}) error {
 
 // DecodeGossip decodes the bytes to the protobuf gossip message provided.
 func (e SszNetworkEncoder) DecodeGossip(b []byte, to interface{}) error {
-	var err error
+	size, err := snappy.DecodedLen(b)
+	if uint64(size) > MaxGossipSize {
```

Comment:
With this here, we can remove the check below as its a duplicate

Reply:
Done, thanks!